core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/leonardo/LeonardoService.scala

```diff
@@ -14,12 +15,18 @@ import org.broadinstitute.dsde.workbench.client.leonardo.model.{
   ListAppResponse,
   ListRuntimeResponse
 }
+import org.broadinstitute.dsde.workbench.model.Notifications.WorkspaceName
 
 import java.util.UUID
 import scala.concurrent.{blocking, ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
-class LeonardoResourceDeletionAction(leonardoDAO: LeonardoDAO)(implicit
```

Comment:
Much of the logic in this class, although oriented around deletion, seemed reusable enough that I wanted to capture it in a dedicated service class so others who are reaching for leonardo APIs can use this and have sane retry and error handling. 

There are still a few instances of using a "bare" Leo DAO in the rawls codebase that should get moved over eventually.